### DISCUSSION TITLE
i18n with fastest-validator

### DISCUSSION THREAD

**Discussion by mtcarvalho** [moleculerjs/moleculer#1083] [Q&A]
I have this method and schema 

```
create: {
			rest: {
				method: "POST",
				path: "/user/deactivated"
			},
			params: {
				user: {
					type: "object",
					props: {
						username: { type: "string", min: 2, pattern: /^[a-zA-Z0-9]+$/ },
						password: { type: "string", min: 6 },
						email: { type: "email" },
						bio: { type: "string", optional: true },
						image: { type: "string", optional: true },
					}
				} 
			},
			...
		}
```

And I have to translate the validation messages to PT-BR using fastest-validator.

I created [this](https://gist.github.com/mtcarvalho/8127a2dca6857a1bfaf0e549f17ad6ee) file to use the messages as I need.

However, I don't know how to use them after importing the custom validator inside my service.

I saw [here](https://github.com/icebob/fastest-validator/issues/38) how to import the translated validation, and [here](https://github.com/moleculerjs/moleculer/blob/05d22d167132057c8ca0d3028fe94850b8d5837f/docs/MIGRATION_GUIDE_0.14.md#use-custom-validator) how to set my custom validator. I tried this code:

```
const { messages } = ('./validation/pt-br')
const Validator = require("fastest-validator");
const validators = {
	'br': new Validator(messages)
}

module.exports = {
	name: "user",
	validator: validators['br'],

	...
}
```

but I don't know if I stay with `params` at the action level.

Maybe I need to create a custom class, like [Joi example](https://moleculer.services/docs/0.14/validating.html#Custom-validator)? If yes, how should I change `params`?

At the end of the day, what I need is: if I send the following body:

```
{
}
```

The response should be:

```
{
    "code": 422,
    "type": "VALIDATION_ERROR",
    "data": [
        {
            "type": "required",
            "message": "O campo 'user' é obrigatório.",
            "field": "user",
            "nodeID": "localhost",
            "action": "user.create"
        }
    ],
    "retryable": false
}
```

And so on, for nested fields as well.

Thanks in advance.

**Comment by saeedtabrizi** [comment]
Hi @mtcarvalho 
I think we need a feature for support context base for any action call validation message localization  support . (similar to i18n-next )
What is your idea @icebob  ?
